### PR TITLE
Updated years datatype to int in 09-plotting

### PR DESCRIPTION
**Pull request by riwoodward**
Following on from Issue #296 I agree with @timothylwarren that the current "plot from dataframe" episode is confusing due to the column headers being a string of the form: `gdpPercap_1962`. At present, the images in the lesson show cropped partly visible x tick labels and when I execute the current commands (below) myself (Matplotlib 2.2.2 & Pandas 0.22.0), the x tick labels are not shown at all:

    data = pandas.read_csv('data/gapminder_gdp_oceania.csv', index_col='country')
    data.loc['Australia'].plot()
    plt.xticks(rotation=90)

Additionally, it's not good practice to plot line graphs where the x-axis is a string representation of a number (here it's ok since the data is evenly spaced, but in general, this isn't a good idea).

Therefore, in this PR I've moved the code from @timothylwarren that extract years from the column headers to the beginning, and updated the dataframe so these values are now integers. 

All code that follows therefore produces better looking plots, which also make more sense since the x-axis now has physical numerical meaning.

Hopefully this adds clarity for learners. If any changes are required, please let me know.

### CODE REVIEW

**Comment by vahtras** [comment]
Thanks! I think the figures look better, but now there are things that are not rendered correctly in the lower part of the page (unless I am doing something wrong).  Please check. For example I get

chop

File: _episodes/09-plotting.md [hunk 1]
```diff
@@ -36,21 +36,27 @@ plt.xlabel('Time (hr)')
 plt.ylabel('Position (km)')
 ~~~
 {: .python}
```
Comment (fmichonneau) [inline _episodes/09-plotting.md]: you need to insert a blank line after `{: .python}`. The language tags should also be changed to `{: .language-python}` to allow for syntax highlighting.

File: _episodes/09-plotting.md [hunk 2]
```diff
@@ -36,21 +36,27 @@ plt.xlabel('Time (hr)')
...
+
+# Extract year from last 4 characters of each column name
+years = data.columns.str.strip('gdpPercap_')
+# Convert year values to integers, saving results back to dataframe
+data.columns = years.astype(int)
+
 data.loc['Australia'].plot()
-plt.xticks(rotation=90)
 ~~~
 {: .python}
```
Comment (fmichonneau) [inline _episodes/09-plotting.md]: same here

File: _episodes/09-plotting.md [hunk 3]
```diff
@@ -59,36 +65,34 @@ plt.xticks(rotation=90)
...
 *   For example, do a bar plot using a fancier style.
 
 ~~~
 plt.style.use('ggplot')
 data.T.plot(kind='bar')
-plt.xticks(rotation=90)
 plt.ylabel('GDP per capita')
 ~~~
 {: .python}
 ![GDP barplot for Australia](../fig/9_gdp_bar.png)
```
Comment (fmichonneau) [inline _episodes/09-plotting.md]: same here

File: _episodes/09-plotting.md [hunk 4]
```diff
@@ -59,36 +65,34 @@ plt.xticks(rotation=90)
...
-
-# Australia data 
+# Get Australia data from dataframe
+years = data.columns
 gdp_australia = data.loc['Australia']
 
-# Plot: 'g--' sets the line style.
 plt.plot(years, gdp_australia, 'g--')
 ~~~
 {: .python}
```
Comment (fmichonneau) [inline _episodes/09-plotting.md]: same here

**Comment by alee** [comment]
Thanks for the PR @riwoodward! I added a commit to fix the formatting. 

In order to view the rendered version you'd need to install Jekyll 3.7 on your machine, or you can try a  [docker-compose / Docker](https://github.com/alee/lesson-infrastructure/tree/docker/docker) setup too.

**Comment by fmichonneau** [comment]
@riwoodward our [lesson-example](https://github.com/swcarpentry/lesson-example) repository has some documentation about the tools we use for our lessons. For an overview on how all the pieces fit together read [episode 2](http://swcarpentry.github.io/lesson-example/02-tooling/index.html) and to learn how to preview locally read [episode 7](http://swcarpentry.github.io/lesson-example/07-checking/index.html). Thanks!

**Pull request merged**